tests/tests_nbde_server_keys_dir.yml

```diff
@@ -0,0 +1,135 @@
+---
+# When using either nbde_server_fetch_keys or nbde_server_deploy_keys,
+# nbde_server_keys_dir needs to be set using an absolute path.
+
+- name: Test nbde_server_keys_dir is properly set
+  hosts: all
+
+  tasks:
+    - name: Test 1 - only nbde_server_fetch_keys set
+      block:
+        - name: With nbde_server_fetch_keys
+          import_role:
+            name: linux-system-roles.nbde_server
+          vars:
+            nbde_server_fetch_keys: yes
+
+        - name: This task should not run
+          debug:
+            msg: Previous task should have failed
+          register: nbde_server_failed
+      rescue:
+        - debug:
+            msg: Test succeeded. Role failed with nbde_server_keys_dir not set
+          register: nbde_server_success
+      always:
+        - fail:
```

Comment:
Does this work?  This is different than the way we usually handle this.  Can you demonstrate that if you make the task `name: With nbde_server_fetch_keys` succeed, the failure case is handled correctly?

Reply:
Yeah, it works: to verify this, just set `nbde_server_keys_dir` to some valid absolute path:
`nbde_server_keys_dir: /tmp/dir`

```
<snip>
TASK [This task should not run]
<snip>
ok: [fedora] => {
    "msg": "Previous task should have failed"
}

TASK [fail] 
<snip>
fatal: [fedora]: FAILED! => {"changed": false, "msg": "Task should have failed with nbde_server_keys_dir not set"}

PLAY RECAP
<snip>
fedora                     : ok=16   changed=3    unreachable=0    failed=1    skipped=5    rescued=0    ignored=0
```

And indeed, this is very clumsy. How would you suggest we handle this?

Reply:
And without changes (i.e. `nbde_server_keys_dir` not set), we detect the issue happened, but the play won't fail and we proceed to the next test:
```
<snip>
TASK [linux-system-roles.nbde_server : Make sure nbde_server_keys_dir is set] 
<snip>
fatal: [fedora]: FAILED! => {"changed": false, "msg": "You need to specify nbde_server_keys_dir and it must be an absolute path\n"}

TASK [debug] <snip>
ok: [fedora] => {
    "msg": "Test succeeded. Role failed with nbde_server_keys_dir not absolute path\n"
}

TASK [fail] <snip>
skipping: [fedora]

PLAY RECAP
<snip>
fedora                     : ok=31   changed=0    unreachable=0    failed=0    skipped=10   rescued=5    ignored=0
```

Reply:
This is how we handle this in other roles. https://github.com/linux-system-roles/kernel_settings/pull/14/files - have a dummy task that does a `fail`, and use `ansible_failed_result` to ensure that the dummy task did not execute

Reply:
Thanks, I will see about using the same pattern here.

Reply:
Updated this one to use the same pattern with `fail` + checking `ansible_failed_result`.